packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -322,56 +336,79 @@ public boolean onRequestPermissionsResult(
     boolean permissionGranted =
         grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED;
 
-    if (requestCode == REQUEST_EXTERNAL_STORAGE_PERMISSION) {
-      if (permissionGranted) {
-        launchPickImageFromGalleryIntent();
-      } else {
-        finishWithSuccess(null);
-      }
-      return true;
-    } else if (requestCode == REQUEST_CAMERA_PERMISSION) {
-      if (permissionGranted) {
-        launchTakeImageWithCameraIntent();
-      } else {
-        finishWithSuccess(null);
-      }
-      return true;
+    switch (requestCode) {
+      case REQUEST_EXTERNAL_IMAGE_STORAGE_PERMISSION:
+        if (permissionGranted) {
+          launchPickImageFromGalleryIntent();
+        }
+        break;
+      case REQUEST_EXTERNAL_VIDEO_STORAGE_PERMISSION:
+        if (permissionGranted) {
+          launchPickVideoFromGalleryIntent();
+        }
+        break;
+      case REQUEST_CAMERA_IMAGE_PERMISSION:
+        if (permissionGranted) {
+          launchTakeImageWithCameraIntent();
+        }
+        break;
+      case REQUEST_CAMERA_VIDEO_PERMISSION:
+        if (permissionGranted) {
+          launchTakeVideoWithCameraIntent();
+        }
+        break;
+      default:
+        return false;
     }
-    return false;
+
+    if (!permissionGranted) {
+      finishWithSuccess(null);
+    }
+
+    return true;
   }
 
   @Override
   public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
-    if (requestCode == REQUEST_CODE_CHOOSE_FROM_GALLERY) {
-      handleChoosePictureResult(resultCode, data);
-      return true;
-    } else if (requestCode == REQUEST_CODE_TAKE_WITH_CAMERA) {
-      handleTakePictureResult(resultCode);
-      return true;
+    switch (requestCode) {
+      case REQUEST_CODE_CHOOSE_IMAGE_FROM_GALLERY:
+        handleChooseMediaResult(resultCode, data, true);
+        break;
+      case REQUEST_CODE_TAKE_IMAGE_WITH_CAMERA:
+        handleCaptureMediaResult(resultCode, true);
+        break;
+      case REQUEST_CODE_CHOOSE_VIDEO_FROM_GALLERY:
+        handleChooseMediaResult(resultCode, data, false);
+        break;
+      case REQUEST_CODE_TAKE_VIDEO_WITH_CAMERA:
+        handleCaptureMediaResult(resultCode, false);
+        break;
+      default:
+        return false;
     }
 
-    return false;
+    return true;
   }
 
-  private void handleChoosePictureResult(int resultCode, Intent data) {
+  private void handleChooseMediaResult(int resultCode, Intent data, boolean isResizable) {
```

Comment:
Please consider breaking this method and the ones below in two to avoid the boolean argument. E.g. `handleChooseImageResult` and `handleChooseVideoResult`. I'd rather suffer a few lines of repeated code between the image vs video cases than having multiple methods whose boolean argument reveal that each does two things.

It doesn't seem unlikely that the image vs video case might drift further apart in the future.